frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1533,21 +1536,27 @@ impl<T: Config> Pallet<T> {
 		<QueuedSolution<T>>::take()
 			.map_or_else(
 				|| {
-					T::Fallback::elect()
-						.map_err(|fe| ElectionError::Fallback(fe))
-						.map(|supports| (supports, ElectionCompute::Fallback))
+					T::Fallback::elect().map_err(|fe| ElectionError::Fallback(fe)).map(|supports| {
+						(supports.clone(), supports.evaluate(), ElectionCompute::Fallback)
+					})
 				},
-				|ReadySolution { supports, compute, .. }| Ok((supports, compute)),
+				|ReadySolution { supports, score, compute }| Ok((supports, score, compute)),
```

Comment:
the `.map_or_else` doesn't make sense here anymore -- let's do: `.or_else(|| -> ReadySolution)` instead?

Reply:
> the `.map_or_else` doesn't make sense here anymore -- let's do: `.or_else(|| -> ReadySolution)` instead?
[or_else()](https://doc.rust-lang.org/std/option/enum.Option.html#method.or_else) and the second argument in it both return `Option<T>` , and the Err value returned by T::Fallback::elect() will be lost when convert it to `Option<T>` .
IMO , it's ok to keep `.map_or_else` .

Reply:
Result also has `or_else`: https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else

Reply:
To rephrase, I don't think mapping from `ReadySolution { x y z }` to `(x, y, z)` is meaningful in any way -- we should just return `ReadySolution { x y z }` and use it further down the line.